src/pybind/mgr/crash/module.py

```diff
@@ -222,22 +239,41 @@ def do_ls(self, cmd, inbuf):
                                '' if 'archived' in c else '*'])
             return 0, table.get_string(), ''
 
-    def do_rm(self, cmd, inbuf):
-        crashid = cmd['id']
+    @CLIReadCommand('crash ls')
+    def do_ls_all(self, format: Optional[str] = None) -> Tuple[int, str, str]:
+        """
+        Show new and archived crash dumps
+        """
+        return self._do_ls(self.crashes.values(), format)
+
+    @CLIReadCommand('crash ls-new')
+    def do_ls_new(self, format: Optional[str] = None) -> Tuple[int, str, str]:
+        """
+        Show new crash dumps
+        """
+        t = [crash for crashid, crash in self.crashes.items()
+             if 'archived' not in crash]
+        return self._do_ls(t, format)
+
+    @CLICommand('crash rm')
+    def do_rm(self, id: str) -> Tuple[int, str, str]:
+        """
+        Remove a saved crash <id>
+        """
+        crashid = id
         if crashid in self.crashes:
             del self.crashes[crashid]
             key = 'crash/%s' % crashid
             self.set_store(key, None)       # removes key
             self._refresh_health_checks()
         return 0, '', ''
 
-    def do_prune(self, cmd, inbuf):
-        keep = cmd['keep']
-        try:
-            keep = int(keep)
-        except ValueError:
-            return errno.EINVAL, '', 'keep argument must be integer'
-
+    @CLICommand('crash prune')
+    @with_crashes
+    def do_prune(self, keep: int) -> Tuple[int, str, str]:
```

Comment:
I'm not sure if this works! `@with_crashes` removes the annotations for the parameters and `@CLICommand` takes the annotations for parameters to compute the arguments for this MON command.

Reply:
yeah, that's why i added this before returning the wrapper.
```py
wrapper.__signature__ = inspect.signature(func)
```

Reply:
And this is needed, independent of what annotations we use for `with_crashes`.